### DISCUSSION TITLE
Can we filter out the blocks that are already fully compressed and then downsample them. This way downsampled blocks will not be compressed.There shouldn't be any other problems with this, right?

### DISCUSSION THREAD

**Discussion by lu-xiansheng** [thanos-io/thanos#7612] [General]
Can we filter out the blocks that are already fully compressed and then downsample them. This way downsampled blocks will not be compressed.There shouldn't be any other problems with this, right?

_Originally posted by @lu-xiansheng in https://github.com/thanos-io/thanos/issues/7334#issuecomment-2274847281_

**Comment by yeya24** [comment]
> fully compressed

Overlapped blocks can trigger vertical compaction even though the block is compacted to the largest block duration

**Reply by lu-xiansheng** [reply]
The block has been compressed, will there be any overlapping blocks?
